types/bun/package.json

```diff
@@ -0,0 +1,42 @@
+{
+    "private": true,
+    "name": "@types/bun",
+    "version": "1.0.9999",
+    "projects": [
+        "https://bun.sh"
+    ],
+    "dependencies": {
+        "@types/node": "*",
```

Comment:
Is this actually how Bun works or is this just for convenience? Because now you're saying that any Bun version has the latest APIs of Node.js

Reply:
By and large, it's how Bun works, but it's true that this is partially aspirational. My previous PR re-declared types for all built-in Node.js modules and we received the opposite feedback (too redundant). There's tradeoffs with both but ultimately Bun aims to be fully Node compatible and IMO our type declarations are not the right place to indicate  deviations from that goal. We document our compatibility here: https://bun.sh/docs/runtime/nodejs-apis

Reply:
Also, if we redeclare these modules, it is virtually impossible to avoid compilation errors if a user has both `@types/bun` and `@types/node` installed simultaneously, which will inevitably happen very frequently. Jake Bailey  rightly expressed this concern here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67275#issuecomment-1793353029

Reply:
This seems acceptable. (afaik all DT packages do this that way, if used)